fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/OrderQueryKeyExpression.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * OrderQueryKeyExpression.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2024 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.query.expressions;
+
+import com.apple.foundationdb.annotation.API;
+import com.apple.foundationdb.record.metadata.expressions.OrderFunctionKeyExpression;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * Handle {@link OrderFunctionKeyExpression} in a query.
+ */
+@API(API.Status.EXPERIMENTAL)
+public class OrderQueryKeyExpression extends QueryKeyExpression {
+
+    public OrderQueryKeyExpression(@Nonnull OrderFunctionKeyExpression keyExpression) {
+        super(keyExpression);
+    }
+
+    /**
+     * Turn a comparison on the inner key (usually a field) into the corresponding comparison on this
+     * ordering key. In addition to encoding the comparand, the direction of the comparison may need to
+     * be reversed if the ordering is (@code DESC).
+     * @param comparison the comparison on the inner key expression
+     * @return a comparison component on this order key expression or {@code null} if not supported
+     */
+    @Nullable
+    public QueryKeyExpressionWithComparison adjustComparison(@Nonnull Comparisons.Comparison comparison) {
+        final boolean inverted = ((OrderFunctionKeyExpression)keyExpression).getDirection().isInverted();
+        Comparisons.Type type = comparison.getType();
+        switch (type) {
+            case EQUALS:
+            case NOT_EQUALS:
+                break;
+            case LESS_THAN:
```

Comment:
Does this also need to do something to handle nulls? For example, if the query comparison is `< "foo"`, I believe this gets translated to a query from `> null, < "foo"`, which would may need to be adjusted if the ordering is contraflows nulls

Reply:
Ah, right, it does sometimes need a second comparison.